crates/astria-conductor/src/executor/mod.rs

```diff
@@ -108,84 +142,94 @@ impl Handle<StateIsInit> {
         &self,
         block: FilteredSequencerBlock,
-    ) -> Result<(), channel::TrySendError<FilteredSequencerBlock>> {
-        self.soft_blocks.try_send(block)
+    ) -> Result<(), SoftTrySendError> {
+        let chan = self.soft_blocks.as_ref().ok_or(SoftTrySendError::NotSet)?;
+        chan.try_send(block)
+            .map_err(|source| SoftTrySendError::Channel {
+                source: Box::new(source),
+            })?;
+        Ok(())
     }
 
-    pub(crate) fn next_expected_firm_height(&mut self) -> SequencerHeight {
-        self.state.borrow_and_update().next_firm_sequencer_height()
+    pub(crate) fn next_expected_firm_sequencer_height(&mut self) -> SequencerHeight {
+        self.state.next_expected_firm_sequencer_height()
     }
 
-    pub(crate) fn next_expected_soft_height(&mut self) -> SequencerHeight {
-        self.state.borrow_and_update().next_soft_sequencer_height()
+    pub(crate) fn next_expected_soft_sequencer_height(&mut self) -> SequencerHeight {
+        self.state.next_expected_soft_sequencer_height()
     }
 
     pub(crate) async fn next_expected_soft_height_if_changed(
         &mut self,
     ) -> Result<SequencerHeight, RecvError> {
-        self.state.changed().await?;
-        Ok(self.state.borrow_and_update().next_soft_sequencer_height())
+        self.state.next_expected_soft_height_if_changed().await
     }
 
     pub(crate) fn rollup_id(&mut self) -> RollupId {
-        self.state.borrow_and_update().rollup_id()
+        self.state.rollup_id()
     }
 
     pub(crate) fn celestia_base_block_height(&mut self) -> CelestiaHeight {
-        self.state.borrow_and_update().celestia_base_block_height()
+        self.state.celestia_base_block_height()
     }
 
     pub(crate) fn celestia_block_variance(&mut self) -> u32 {
-        self.state.borrow_and_update().celestia_block_variance()
+        self.state.celestia_block_variance()
     }
 }
 
 pub(crate) struct Executor {
-    firm_blocks: mpsc::Receiver<ReconstructedBlock>,
-    soft_blocks: channel::Receiver<FilteredSequencerBlock>,
+    firm_blocks: Option<mpsc::Receiver<ReconstructedBlock>>,
+    soft_blocks: Option<channel::Receiver<FilteredSequencerBlock>>,
 
     /// Token to listen for Conductor being shut down.
     shutdown: CancellationToken,
 
     rollup_address: tonic::transport::Uri,
 
-    /// Tracks SOFT and FIRM on the execution chain
-    state: watch::Sender<State>,
-
-    // If set, the executor will take into account the spread between firm
-    // and soft commitments when executing blocks.
-    consider_commitment_spread: bool,
+    /// Tracks the status of the execution chain.
+    state: StateSender,
 
     /// Tracks executed blocks as soft commitments.
     ///
     /// Required to mark firm blocks received from celestia as executed
     /// without re-executing on top of the rollup node on top of the rollup node..
-    blocks_pending_finalization: HashMap<[u8; 32], Block>,
+    blocks_pending_finalization: HashMap<u32, Block>,
```

Comment:
is this rollup_block_height to block?

Reply:
Would be good to have a comment above on what the height is for next person through.

Reply:
Yes, this was confusing. Changed to:
```
    /// Tracks rollup blocks by their rollup block numbers.
    ///
    /// Required to mark firm blocks received from celestia as executed
    /// without re-executing on top of the rollup node.
```